server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -235,7 +235,7 @@ protected void doInternalExecute(Task task, BulkRequest bulkRequest, String exec
             }
 
             if (actionRequest instanceof IndexRequest ir) {
-                ir.checkAutoIdWithOpTypeCreateSupportedByVersion(minNodeVersion);
+                ir.checkAutoIdWithOpTypeCreateSupportedByVersion(minNodeVersion.transportVersion);
```

Comment:
`minNodeVersion` will need to be migrated separately, its used in several other places